### ISSUE TITLE
Ingested not working

### ISSUE THREAD

**Issue by ismailkalolwala** [dainok/netdoc#76]
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]: An exception occurred: `TypeError: Device() got unexpected keyword arguments: 'device_role_id'`
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]: ```
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]: Traceback (most recent call last):
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:   File "/opt/netbox/netbox/extras/scripts.py", line 504, in _run_script
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:     script.output = script.run(data=data, commit=commit)
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:   File "/opt/netbox/netbox/scripts/netdoc_scripts.py", line 353, in run
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:     log_ingest(log)
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:   File "/opt/netbox/venv/lib/python3.10/site-packages/netdoc/utils.py", line 578, in log_ingest
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:     module.ingest(log)
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:   File "/opt/netbox/venv/lib/python3.10/site-packages/netdoc/ingestors/netmiko_cisco_ios_hostname.py", line 36, in ingest
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:     device_o = device.create(**data)
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:   File "/opt/netbox/venv/lib/python3.10/site-packages/netdoc/schemas/device.py", line 108, in create
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:     obj = utils.object_create(Device, **kwargs)
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:   File "/opt/netbox/venv/lib/python3.10/site-packages/netdoc/utils.py", line 1055, in object_create
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:     return model_o.objects.create(**kwargs)
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:   File "/opt/netbox/venv/lib/python3.10/site-packages/django/db/models/manager.py", line 87, in manager_method
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:     return getattr(self.get_queryset(), name)(*args, **kwargs)
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:   File "/opt/netbox/venv/lib/python3.10/site-packages/django/db/models/query.py", line 656, in create
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:     obj = self.model(**kwargs)
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:   File "/opt/netbox/netbox/utilities/tracking.py", line 41, in __init__
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:     super().__init__(*args, **kwargs)
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:   File "/opt/netbox/venv/lib/python3.10/site-packages/django/db/models/base.py", line 567, in __init__
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]:     raise TypeError(
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]: TypeError: Device() got unexpected keyword arguments: 'device_role_id'
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]: ```
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]: Exception raised during script execution: Device() got unexpected keyword arguments: 'device_role_id'
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]: 13:42:54 default: Job OK (1b80535e-28ed-4600-8627-bd82c3eb29aa)
Sep 28 13:42:54 dc-netbox-srv01 python3[13861]: 13:42:54 Result is kept for 500 seconds

**Comment by ismailkalolwala** [comment]
![image](https://github.com/dainok/netdoc/assets/145698133/571f1301-993f-4060-b606-751f739d6158)

**Comment by ismailkalolwala** [comment]
I did a reinstallation of the NETDOC and now it is showing me this 

![image](https://github.com/dainok/netdoc/assets/145698133/a70b3fd2-ec67-463b-a072-50d09b0ac262)

**Comment by dainok** [comment]
Seems that the migration to Netbox 3.5 brings a lot of unexpected errors, even if I tested NetDoc on my lab. I need to make additional tests.
If you are hurry, please use NetDoc 0.10.x with NetBox 3.4.10.
Stay tuned

**Comment by ismailkalolwala** [comment]
![image](https://github.com/dainok/netdoc/assets/145698133/4234c236-4598-47c3-b8e6-6cb8364f3829)

**Comment by ismailkalolwala** [comment]
i can wait. Let me know once your testing is completed. and then i can roll it up in my environment.

**Comment by dainok** [comment]
https://github.com/dainok/netdoc/pull/78 should fix the issue. Please upgrade to 3.5.4 and let me know if anything changes.

**Comment by elixdreamer** [comment]
Facing similar issue since i'm deploying a new Netbox environment in version 3.6.3.

**Comment by dainok** [comment]
@elixdreamer please post the error and how you get into it

**Comment by elixdreamer** [comment]
Had to add the report and script manually since these aren't auto imported when installing the plugin.
Scripts downloaded from this GitHub

Once adding a device a discover is done. This discover is successful.
Checking later the ingest script does fail with the below error:
![image](https://github.com/dainok/netdoc/assets/12197783/80c72f3c-dc36-45fa-a53d-58af8949e78c)

**Comment by dainok** [comment]
The error is different, please see https://github.com/dainok/netdoc/issues/74

**Comment by elixdreamer** [comment]
Can't see the differences between #74 and this issue...

**Comment by dainok** [comment]
@elixdreamer my mistake, it is duplicated, sorry.

**Comment by dainok** [comment]
@elixdreamer which netbox version are you using? device_role_id exists on NetBox 3.5 and is renamed to role_id into Netbox 3.6 which is not supported by NetDoc (not yet).

**Comment by dainok** [comment]
With Netbox 3.6 you will get device_role_id issue. Currenlty I'm supporting NetBox 3.5 only.

**Comment by proebstle** [comment]
Hi,
   
  
    
   
  
   now with version 3.5 no issue.
   
  
   is it possible to add Aroba CX Switches to Discoverable devices?
   
  
    
   
  
   Best regards from Germany
   
  
   Gilbert
   
  
    
   
  
    
   
  
    
   
  
   /opt/ntc-templates/ntc_templates/templates/
   
  
    
   
  
   aruba_aoscx_show_aaa_authentication_port-access_interface_all_client-status.textfsm aruba_aoscx_show_ip_route_all-vrfs.textfsm
   aruba_aoscx_show_arp_all-vrfs.textfsm
   
  
   aruba_aoscx_show_lldp_neighbors-info_detail.textfsm
   aruba_aoscx_show_bfd_all-vrfs.textfsm
   
  
   aruba_aoscx_show_mac-address-table.textfsm
   aruba_aoscx_show_bgp_all_all-vrfs_summary.textfsm aruba_aoscx_show_ntp_associations.textfsm
   aruba_aoscx_show_bgp_all-vrfs_all_summary.textfsm
   
  
   aruba_aoscx_show_system.textfsm
   aruba_aoscx_show_interface_dom_detail.textfsm
   
  
   aruba_aoscx_show_vsf_detail.textfsm
   aruba_aoscx_show_interface.textfsm

**Comment by dainok** [comment]
@proebstle yes, it will require a little bit.
Can you prepare a lab and give me the outputs?
See also https://github.com/dainok/netdoc/wiki/Developing-NetDoc#adding-support-for-new-devices

**Comment by proebstle** [comment]
Hi,
   
  
    
   
  
   we have more Aruba CX switches and a Aruba CX VSX cluter in out environment.
   
  
   I can make outputs and tests for you.
   
  
    
   
  
   Best regards
   
  
   Gilbert
   
  
    
   
   
   
    Andrea Dainese ***@***.***> hat am 05.10.2023 16:47 CEST geschrieben:
    
   
     
    
   
     
    
   
    @proebstle yes, it will require a little bit.
    
   
    Can you prepare a lab and give me the outputs?
    
   
    See also https://github.com/dainok/netdoc/wiki/Developing-NetDoc#adding-support-for-new-devices

**Comment by dainok** [comment]
In case open a new issue to track improvements.
